src/hapi/pipelines/database/operational_presence.py

```diff
@@ -87,26 +81,13 @@ def populate(self):
                                 reference_period_start=reference_period_start,
                                 reference_period_end=reference_period_end,
                             )
-
-                        if sector_name_index:
-                            sector_name = values[sector_name_index][
-                                admin_code
-                            ][i]
-                            sector_code = self._sector.get_sector_info(
-                                sector_name, "name"
-                            )
-                        else:
-                            sector_code = values[sector_code_index][
-                                admin_code
-                            ][i]
-                            sector_name = self._sector.get_sector_info(
-                                sector_code, "code"
-                            )
-                        if sector_code == "" or sector_name == "":
-                            # TODO: Fuzzy match sectors (HAPI-193)
-                            logger.error(
-                                f"Sector {sector_name, sector_code} not in table"
-                            )
+                        sector = None
+                        sector_code = None
+                        if sector_index:
+                            sector = values[sector_index][admin_code][i]
+                            sector_code = self._sector.get_sector_code(sector)
+                        if sector and not sector_code:
```

Comment:
Why do you also check for `sector` in this line - could it happen that `sector` is also None?

Reply:
It could!  Some datasets don't include sector at all.  Actually, in those cases, what would we fill in instead of null in the OP table?  Maybe too early to figure out.